src/core/packet.lua

```diff
@@ -20,10 +18,40 @@ local header_size = 8
 local max_payload = tonumber(C.PACKET_PAYLOAD_SIZE)
 
 -- Freelist containing empty packets ready for use.
+
+ffi.cdef[[
+struct freelist {
+    int nfree, max;
+    struct packet *list[?];
+};
+]]
```

Comment:
I did some quick basic benchmark testing and got slightly better results with:

```
ffi.cdef[[
struct freelist {
    uint64_t nfree, max;
    __attribute__((aligned(64)))
    struct packet *list[?];
};
]]
```

I understand that it is probably annoying to insist here - but this and `struct packet` are the most used (at runtime) data structures in the project, so I believe every detail here is important. So if this is going upstream, IMHO we should look at it very carefully.